src/providers/storage/azureBlobStorage.ts

```diff
@@ -138,13 +141,41 @@ export class AzureCloudStorageService implements IStorageProvider {
             path = path ? [this.options.containerName, path].join("/") : this.options.containerName;
         }
         const files = await this.listFiles(path);
-        return files.map((blobPath) => AssetService.createAssetFromFilePath(blobPath));
+        const result: IAsset[] = [];
+        for (const key of Object.keys(files.entries)) {
+            const url = this.getUrl(files.entries[key].name);
+            const asset = AssetService.createAssetFromFilePath(url);
+            if (asset.type === AssetType.Image || asset.type === AssetType.Video) {
+                result.push(asset);
+            }
+        }
+        return result;
     }
 
     private getService() {
         return AzureStorageBlob.createBlobService(this.options.connectionString);
     }
 
+    private getUrl(blobName: string) {
```

Comment:
Is this a static call via the library? Or does it make an HTTP request to figure it out?

Reply:
I believe it's a static call